src/pages/Signup.svelte

```diff
@@ -0,0 +1,116 @@
+<script>
+  import axios from "axios";
+  import { user } from "../stores/store";
+  import { push, link } from "svelte-spa-router";
+
+  import Button from "../components/Button.svelte";
+  import Input from "../components/Input.svelte";
+  import { LOGIN_PAGE } from "../constants/routes";
+
+  let email;
+  let username;
+  let password;
+
+  async function signup() {
+    try {
+      await axios.post("/api/auth/signup", { email, username, password });
+      $user = data.user;
+      push("/home");
+    } catch (e) {
+      if (e) {
+        console.log(e);
```

Comment:
.

Reply:
I think we don't need if inside a catch block.